instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py

```diff
@@ -119,7 +119,14 @@ def __init__(
         self._register_event_listener(engine, "checkout", self._pool_checkout)
 
     def _get_pool_name(self):
-        return self.engine.pool.logging_name or ""
+        if self.engine.pool.logging_name is not None:
+            return self.engine.pool.logging_name
+
+        drivername = self.engine.url.drivername or ""
+        host = self.engine.url.host or ""
+        port = self.engine.url.port or ""
+        database = self.engine.url.database or ""
+        return f"{drivername}://{host}:{port}/{database}"
```

Comment:
Is this the recommendation? I see it says connection string now

Reply:
Indeed but the connection string is not implemented for sqlalchemy.
The way the connection string is calculated for js-contrib is like this:

https://github.com/open-telemetry/opentelemetry-js-contrib/blob/69d1f17da0914fe14dc5d48512e8ae829267f605/plugins/node/opentelemetry-instrumentation-pg/src/utils.ts#L97

Or this one:
https://github.com/open-telemetry/opentelemetry-js-contrib/blob/69d1f17da0914fe14dc5d48512e8ae829267f605/plugins/node/opentelemetry-instrumentation-mysql/src/utils.ts#L56

I can add a get_connection_string function to return this string, It will be more clearly

Reply:
Fixed, way better
thanks for the comment

Reply:
https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.engine.URL 👍